### PR TITLE
Making sure summarizer does not auto start and adding consistency checks

### PR DESCRIPTION
**Pull request by NicholasCouri**
I realized there were sometimes 2 summarizers running at the same time and it is due to the fact the default behavior, unless we explicitly disable heuristics, when creating a container. 

Also, I'm trying to enforce the interruption of the test whenever we see an inconsistency trying to get row or col from the matrix in case it happens where the rowId or colId should be.  It ends up that it is happening consistently on the summarizer runs. 

I will keep on investigating but wanted to get this in ASAP so we can catch sooner.

### CODE REVIEW

File: packages/framework/collabspaces/src/collabSpaces.ts [hunk 1]
```diff
@@ -590,18 +590,21 @@ export class CollabSpacesRuntime
 		const col = colArg + 1;
 		assert(row > 0 && col > 0, "arguments");
 		const cellValue = this.matrix.getCell(row, col);
+		const rowId = this.matrix.getCell(row, 0) as unknown as uuidType;
+		const colId = this.matrix.getCell(0, col) as unknown as uuidType;
```
Comment (vladsud) [inline packages/framework/collabspaces/src/collabSpaces.ts:594]: why did you move this? This is not performant (in case we have no value). Is this for extra validation?

File: packages/framework/collabspaces/src/collabSpaces.ts [hunk 2]
```diff
@@ -761,7 +764,11 @@ export class CollabSpacesRuntime
 
 		let row;
 		for (row = 1; row < rowCount; row++) {
-			if (this.areEqualUuid(this.matrix.getCell(row, 0) as unknown as uuidType, rowId)) {
+			const uuidTypeVal = this.matrix.getCell(row, 0) as unknown as uuidType;
+			if (uuidTypeVal === undefined) {
+				throw new Error(`uuidTypeVal is undefined for row ${row}`);
+			}
```
Comment (vladsud) [inline packages/framework/collabspaces/src/collabSpaces.ts:770]: Maybe move this check into areEqualUuid()? You would lose "row" vs. "column" text verbatim - not sure if it matters.

File: packages/framework/collabspaces/src/test/collabSpaces.spec.ts [hunk 3]
```diff
@@ -60,11 +61,20 @@ describe("Temporal Collab Spaces", () => {
 	let loader: IHostLoader | undefined;
 	let seed: number;
 
+	const summaryOptionsToDisableHeuristics: ISummaryRuntimeOptions = {
+		summaryConfigOverrides: {
+			state: "disableHeuristics",
```
Comment (vladsud) [inline packages/framework/collabspaces/src/test/collabSpaces.spec.ts:66]: Great call out RE default summarizer! I bet that's what was holding MSNs!
I'm curious if perf improves simply because MSN can move more freely.
If it is, it would be an interesting option to keep to demonstrate perf problems.

**Review by vladsud — APPROVED** [review]

File: packages/framework/collabspaces/src/collabSpaces.ts (hunk 1 above)
Comment (NicholasCouri) [reply packages/framework/collabspaces/src/collabSpaces.ts:594]: Yes, for extra validation in case we get a cell that is empty, I still want to validate!

File: packages/framework/collabspaces/src/collabSpaces.ts (hunk 2 above)
Comment (NicholasCouri) [reply packages/framework/collabspaces/src/collabSpaces.ts:770]: What I want here is, independently of the row we want to find, is to find inconsistent cells on the matrix row/column. Until we get to the row we want,  I want to capture regressions as soon as possible.
 Does it help ?
